esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2490,32 +2490,37 @@ bool CFileSprayEx::onCopy(IEspContext &context, IEspCopy &req, IEspCopyResponse
         StringBuffer fileMask;
         constructFileMask(destTitle.str(), fileMask);
 
+        //Set the udesc and foreigndali
+        Owned<INode> foreigndali;
+        StringBuffer user, passwd;
         const char* srcDali = req.getSourceDali();
-        bool supercopy = req.getSuperCopy();
-        if (supercopy)
+        if (srcDali)
         {
-            StringBuffer user, passwd;
-            context.getUserID(user);
-            context.getPassword(passwd);
-            StringBuffer u(user);
-            StringBuffer p(passwd);
-            Owned<INode> foreigndali;
-            if (srcDali)
+            SocketEndpoint ep(srcDali);
+            foreigndali.setown(createINode(ep));
+            const char* srcu = req.getSrcusername();
+            if (!isEmptyString(srcu))
             {
-                SocketEndpoint ep(srcDali);
-                foreigndali.setown(createINode(ep));
-                const char* srcu = req.getSrcusername();
-                if(srcu && *srcu)
-                {
-                    u.clear().append(srcu);
-                    p.clear().append(req.getSrcpassword());
-                }
+                user.set(srcu);
+                passwd.set(req.getSrcpassword());
             }
-            Owned<IUserDescriptor> udesc=createUserDescriptor();
-            udesc->set(u.str(),p.str());
-            if (!queryDistributedFileDirectory().isSuperFile(srcname,udesc,foreigndali))
+        }
+        if (user.isEmpty())
+        {
+            context.getUserID(user);
+            context.getPassword(passwd);
+        }
+        Owned<IUserDescriptor> udesc=createUserDescriptor();
+        udesc->set(user, passwd);
+
+        bool supercopy = req.getSuperCopy();
+        if (supercopy)
+        {
+            if (!queryDistributedFileDirectory().isSuperFile(srcname, udesc, foreigndali))
                 supercopy = false;
         }
+        else if (queryDistributedFileDirectory().isSuperIndexFile(srcname, udesc, foreigndali))
+            supercopy = true;
```

Comment:
@wangkx practically those two lines are ok, but theoretically I think  they are wrong because:
- it changes the meaning/usage of 'asSuper' parameter behind the scenes
- it is entirely disable my changes in dfu server because it always force the superkey/superindex copy as supercopy 
(- and finally in educational point of view, it prevents the user/ECL programmer faces to their mistake)

We can start a debate about this changes and about it is good if our platform are cleverer than its users. There has a sense to copy a superfile as a superfile or a simple logical file, but there isn't with a superkey. So, I think in this case we should separate to copy a superfile from copy a superkey. 

If I remove the two lines above from your PR code then the  supercopy.ecl finish well. (With this I remove the main part of your PR as well, sorry)

I think this PR is  a duplicate of the already merged  PR #12606: HPCC-22124 File services permits and defaults to copying a super index into a invalid single index with multiple TLK's by AttilaVamos in hpcc-systems/HPCC-Platform on GitHub. 
Perhaps we should ask an architect.

Reply:
I wish this PR is a duplicate of https://github.com/hpcc-systems/HPCC-Platform/pull/12606. But, I found that https://github.com/hpcc-systems/HPCC-Platform/pull/12606 may not be good enough for ESP. When a user copies a super index without the super file option using ESP, the existing ESP code just creates a job for DFU server. With https://github.com/hpcc-systems/HPCC-Platform/pull/12606, dfuserver may log an error. But, if the user does not read the log, the user does not know what happens. 
I was considering two options for this PR. 1. Throw an exception when a user copies a super index without the super file option. 2. Force the super file option when a user copies a super index. I am choosing the second option because ESP does the similar thing in line 2516-2517 in the existing ws_fsService.cpp. @jakesmith please comment.

Reply:
@wangkx I think the parent JIRA (https://track.hpccsystems.com/browse/HPCC-22139) of this means it would be enough if the ECLWatch  prevents (on or immediately after the copy dialog) the user to trying copy superkey without using  'asSuper' parameter. 

Any other case, if a copy (workunit) fails (the https://github.com/hpcc-systems/HPCC-Platform/pull/12606 throws an exception) and the user/ECL programmer doesn't aware of it /check the logs/results, then I don't know hat we can/should do.

Reply:
@AttilaVamos an ESP service may be called without ECLWatch UI. For example, a user may send a SOAP call to ESP.

Reply:
@wangkx  Yes, but every time the copy goes through DFU server (if I know well), where all necessary checks are already implemented.

Reply:
@AttilaVamos The copy does go through DFU server and DFU server does throw an exception. But, I do not think that most of ECL programmers will check DFU server log even if the copy fails.

Reply:
@wangkx as I mentioned earlier, we can start a debate about...
@jakesmith what is your opinion.

Reply:
I've changed my mind..
Eclwatch users might be expected to look at an index and choose appropriate options (by Eclwatch now suppresses the invalid options).
Other users (SOAP and ECL) who are asking to copy an index, generally have generic code and cannot know (and do not are). This option should be irrelevant (not even an option) for them when copying an index. It is has 1 answer and it can never validly be anything else.
You could argue there should be a separate service call or a separate fileservices call (e.g. CopyIndex), but using std. Copy it just means some options do not apply (make any sense), supercopy is one of them.

If we make it fire an error when they pick badly, then they'll be nasty conditional code reading and testing to set appropriate setting where only 1 answer is ever correct for indexes.
We already ignore supercopy=true if the file isn't a superfile, which also seems reasonable.

So I think it should just ignore supercopy=false if a superindex, i.e. perform as if supercopy=true.
i.e. as @wangkx has done.